backend/corpora/lambdas/api/v1/curation/collections/actions.py

```diff
@@ -18,23 +18,28 @@ def get(visibility: str, token_info: dict, curator: str = None):
     @return: Response
     """
     filters = [DbCollection.tombstone == False]  # noqa
-
-    if visibility == CollectionVisibility.PRIVATE.name and not token_info:
-        raise UnauthorizedError()
-    elif visibility:
+    if visibility:
         filters.append(DbCollection.visibility == getattr(CollectionVisibility, visibility))
+        if visibility == CollectionVisibility.PRIVATE.name:
+            if not token_info:
+                raise ForbiddenHTTPException(detail="Not authorized to query for PRIVATE collection.")
+            else:
+                result = owner_or_allowed(token_info)
+                if result:
+                    filters.append(DbCollection.owner == result)
 
-    if curator and not is_super_curator(token_info):
-        raise UnauthorizedError()
-    elif curator:  # user want collections from a specific curator
-        filters.append(DbCollection.curator_name == curator)
+    if curator:
+        if not is_super_curator(token_info):
+            raise ForbiddenHTTPException(detail="Not authorized to use the curator query parameter.")
+        else:
+            filters.append(DbCollection.curator_name == curator)
 
     db_session = g.db_session
     resp_collections = []
     for collection in db_session.query(DbCollection).filter(*filters).all():
         resp_collection = collection.to_dict_keep(EntityColumns.columns_for_collections)
```

Comment:
suggestion, non-blocking: I think for a 'reshape_for_curation_api' function, it makes sense to refactor all the recurring reshaping logic (to_dict_keep, add_collection_level_processing_status) into that call. It taking in the collection object from the DB and returning a resp_collection (rather than being in-place) also seems like a better pattern to me.

Reply:
Great idea.